### PR TITLE
Conformance tester: Verify reconciliaton success

### LINKED ISSUE #3417: Upgrade test: Verify reconciliation was done prior to running conformance tests
**User Story**

Currently, our upgrade tests:

1. Deploy old Kubermatic
1. Create a Cluster
1. Run conformance tests
1. Upgrade conformance tests
1. Deploy new Kubermatic

This has the issue that we don't wait for the new Kubermatic to actually reconcile the cluster. If it fails, we will never notice and basically just run the tests again against the same, unchanged cluster.

**Acceptance criteria**

* Upgrade tests wait for all Kubermatic controller to successfully reconcile the cluster before running the conformance tests the second time
* To achieve this, all our controllers must emit a metric, telling the version with which they last successfully reconciled: `lastSuccessfullSyncTimestamp{controller_name: string, reqAfter: bool, kubermatic_version: string, cluster_name: string}`
* This includes all controllers in the Kubermatic controller manager and in the usercluster controller manager
* After upgrading Kubermatic, the conformance tester waits for this metric to appear for all controllers
* After this metric appeared, the conformance tester waits for all Deployments and Statefulsets to have `replicas == .status.updatedReplicas` and for all Pods to be `Ready` to ensure all Pods were successfully updated

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Makes the conformance tester wait fro reconciliation success prior to doing anything with the cluster.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3417 

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: api/cmd/conformance-tests/runner.go [hunk 1]
```diff
@@ -638,11 +661,43 @@ func (r *testRunner) executeGinkgoRunWithRetries(log *zap.SugaredLogger, run *gi
...
+			return false, nil
+		}
+		for _, nodeDeployment := range resp.Payload {
+			existingReplicas += int(*nodeDeployment.Spec.Replicas)
+		}
+		return true, nil
+	}); err != nil {
+		return fmt.Errorf("failed to get existing NodeDeployments: %v", err)
+	}
+	log.Info("Found %d pre-existing node replicas", existingReplicas)
```
Comment (xrstf) [inline api/cmd/conformance-tests/runner.go]: Infof

**Comment by alvaroaleman** [comment]
@xrstf updated, PTAL

**Pull request merged**